diff --git a/include/pci_ids/radeonsi_pci_ids.h b/include/pci_ids/radeonsi_pci_ids.h
index 8c4567e8483..9f0d596f4ee 100644
--- a/include/pci_ids/radeonsi_pci_ids.h
+++ b/include/pci_ids/radeonsi_pci_ids.h
@@ -145,6 +145,11 @@ CHIPSET(0x131B, KAVERI)
 CHIPSET(0x131C, KAVERI)
 CHIPSET(0x131D, KAVERI)
 
+CHIPSET(0x9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS)
+
 CHIPSET(0x67A0, HAWAII)
 CHIPSET(0x67A1, HAWAII)
 CHIPSET(0x67A2, HAWAII)
diff --git a/src/amd/addrlib/src/amdgpu_asic_addr.h b/src/amd/addrlib/src/amdgpu_asic_addr.h
index 88dcfbdc44b..767b589ea98 100644
--- a/src/amd/addrlib/src/amdgpu_asic_addr.h
+++ b/src/amd/addrlib/src/amdgpu_asic_addr.h
@@ -71,7 +71,9 @@
 #define AMDGPU_HAWAII_RANGE     0x28, 0x3C
 
 #define AMDGPU_SPECTRE_RANGE    0x01, 0x41
-#define AMDGPU_SPOOKY_RANGE     0x41, 0x81
+#define AMDGPU_SPOOKY_RANGE     0x41, 0x61
+#define AMDGPU_STARSHA_RANGE    0x61, 0x71
+#define AMDGPU_STARSHP_RANGE    0x71, 0x81
 #define AMDGPU_KALINDI_RANGE    0x81, 0xA1
 #define AMDGPU_GODAVARI_RANGE   0xA1, 0xFF
 
@@ -123,6 +125,8 @@
 
 #define ASICREV_IS_SPECTRE(r)          ASICREV_IS(r, SPECTRE)
 #define ASICREV_IS_SPOOKY(r)           ASICREV_IS(r, SPOOKY)
+#define ASICREV_IS_STARSHA(r)           ASICREV_IS(r, STARSHA)
+#define ASICREV_IS_STARSHP(r)           ASICREV_IS(r, STARSHP)
 #define ASICREV_IS_KALINDI(r)          ASICREV_IS(r, KALINDI)
 #define ASICREV_IS_KALINDI_GODAVARI(r) ASICREV_IS(r, GODAVARI)
 
diff --git a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
index 888c6778034..dfd6197b039 100644
--- a/src/amd/common/ac_gpu_info.c
+++ b/src/amd/common/ac_gpu_info.c
@@ -510,6 +510,8 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    case FAMILY_KV:
       identify_chip2(SPECTRE, KAVERI);
       identify_chip2(SPOOKY, KAVERI);
+      identify_chip2(STARSHA, LIVERPOOL);
+      identify_chip2(STARSHP, GLADIUS);
       identify_chip2(KALINDI, KABINI);
       identify_chip2(GODAVARI, KABINI);
       break;
@@ -645,7 +647,7 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    info->pa_sc_tile_steering_override = device_info.pa_sc_tile_steering_override;
    info->max_render_backends = amdinfo->rb_pipes;
    /* The value returned by the kernel driver was wrong. */
-   if (info->family == CHIP_KAVERI)
+   if (info->family == CHIP_KAVERI || info->family == CHIP_LIVERPOOL)
       info->max_render_backends = 2;
 
    /* Guess the number of enabled SEs because the kernel doesn't tell us. */
@@ -701,6 +703,8 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    case CHIP_ICELAND:
    case CHIP_CARRIZO:
    case CHIP_FIJI:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_POLARIS12:
    case CHIP_VEGAM:
       info->l2_cache_size = info->num_tcc_blocks * 128 * 1024;
@@ -1232,6 +1236,8 @@ int ac_get_gs_table_depth(enum chip_class chip_class, enum radeon_family family)
    case CHIP_TONGA:
    case CHIP_FIJI:
    case CHIP_POLARIS10:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_POLARIS11:
    case CHIP_POLARIS12:
    case CHIP_VEGAM:
@@ -1297,6 +1303,14 @@ void ac_get_raster_config(struct radeon_info *info, uint32_t *raster_config_p,
       raster_config = 0x3a00161a;
       raster_config_1 = 0x0000002e;
       break;
+   case CHIP_LIVERPOOL:
+	raster_config = 0x2a00161a;
+	raster_config_1 = 0x00000000;
+	break;
+   case CHIP_GLADIUS:
+	raster_config = 0x2a00161a;
+	raster_config_1 = 0x0000002e;
+	break;
    default:
       fprintf(stderr, "ac: Unknown GPU, using 0 for raster_config\n");
       raster_config = 0x00000000;
diff --git a/src/amd/common/amd_family.h b/src/amd/common/amd_family.h
index 31ec0fb8aab..c4de37947df 100644
--- a/src/amd/common/amd_family.h
+++ b/src/amd/common/amd_family.h
@@ -86,6 +86,8 @@ enum radeon_family
    CHIP_HAINAN,
    CHIP_BONAIRE, /* GFX7 (Sea Islands) */
    CHIP_KAVERI,
+   CHIP_LIVERPOOL,
+   CHIP_GLADIUS,
    CHIP_KABINI,
    CHIP_HAWAII,
    CHIP_TONGA, /* GFX8 (Volcanic Islands & Polaris) */
diff --git a/src/amd/common/amd_family.c b/src/amd/common/amd_family.c
index 801cce1dbe2..eb041650b72 100644
--- a/src/amd/common/amd_family.c
+++ b/src/amd/common/amd_family.c
@@ -41,6 +41,8 @@ const char *ac_get_family_name(enum radeon_family family)
    case CHIP_HAINAN:
       return "hainan";
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
       return "bonaire";
    case CHIP_KABINI:
       return "kabini";
diff --git a/src/amd/llvm/ac_llvm_util.c b/src/amd/llvm/ac_llvm_util.c
index b820a5577a3..fc8b249e580 100644
--- a/src/amd/llvm/ac_llvm_util.c
+++ b/src/amd/llvm/ac_llvm_util.c
@@ -132,6 +132,8 @@ const char *ac_get_llvm_processor_name(enum radeon_family family)
    case CHIP_HAINAN:
       return "hainan";
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
       return "bonaire";
    case CHIP_KABINI:
       return "kabini";
diff --git a/src/amd/vulkan/radv_pipeline.c b/src/amd/vulkan/radv_pipeline.c
index 71b2aaca605..383e383f43a 100644
--- a/src/amd/vulkan/radv_pipeline.c
+++ b/src/amd/vulkan/radv_pipeline.c
@@ -1452,7 +1452,8 @@ radv_compute_ia_multi_vgt_param_helpers(struct radv_pipeline *pipeline)
       /* Bug with tessellation and GS on Bonaire and older 2 SE chips. */
       if ((device->physical_device->rad_info.family == CHIP_TAHITI ||
            device->physical_device->rad_info.family == CHIP_PITCAIRN ||
-           device->physical_device->rad_info.family == CHIP_BONAIRE) &&
+           device->physical_device->rad_info.family == CHIP_BONAIRE ||
+	   device->physical_device->rad_info.family == CHIP_LIVERPOOL) /* ??*/ &&
           radv_pipeline_has_gs(pipeline))
          ia_multi_vgt_param.partial_vs_wave = true;
       /* Needed for 028B6C_DISTRIBUTION_MODE != 0 */
diff --git a/src/amd/vulkan/winsys/null/radv_null_winsys.c b/src/amd/vulkan/winsys/null/radv_null_winsys.c
index 6afd02e8d6e..55969131f9b 100644
--- a/src/amd/vulkan/winsys/null/radv_null_winsys.c
+++ b/src/amd/vulkan/winsys/null/radv_null_winsys.c
@@ -43,6 +43,10 @@ static const struct {
    [CHIP_OLAND] = {0x6060, 2, true},
    [CHIP_HAINAN] = {0x6660, 2, true},
    [CHIP_BONAIRE] = {0x6640, 4, true},
+   [CHIP_LIVERPOOL] = {0x9920, 8, true},
+   [CHIP_LIVERPOOL] = {0x9922, 8, true},
+   [CHIP_LIVERPOOL] = {0x9923, 8, true},
+   [CHIP_GLADIUS] = {0x9924, 16, true},
    [CHIP_KAVERI] = {0x1304, 2, false},
    [CHIP_KABINI] = {0x9830, 2, false},
    [CHIP_HAWAII] = {0x67A0, 16, true},
diff --git a/src/gallium/drivers/radeonsi/si_state_draw.cpp b/src/gallium/drivers/radeonsi/si_state_draw.cpp
index 63644d02925..9c300e6d481 100644
--- a/src/gallium/drivers/radeonsi/si_state_draw.cpp
+++ b/src/gallium/drivers/radeonsi/si_state_draw.cpp
@@ -480,7 +480,8 @@ static unsigned si_get_init_multi_vgt_param(struct si_screen *sscreen, union si_
 
       /* Bug with tessellation and GS on Bonaire and older 2 SE chips. */
       if ((sscreen->info.family == CHIP_TAHITI || sscreen->info.family == CHIP_PITCAIRN ||
-           sscreen->info.family == CHIP_BONAIRE) &&
+           sscreen->info.family == CHIP_BONAIRE ||
+	   sscreen->info.family == CHIP_LIVERPOOL) /* ? */ &&
           key->u.uses_gs)
          partial_vs_wave = true;
 
diff --git a/src/gallium/targets/d3dadapter9/description.c b/src/gallium/targets/d3dadapter9/description.c
index a3e4cd61770..09104f49101 100644
--- a/src/gallium/targets/d3dadapter9/description.c
+++ b/src/gallium/targets/d3dadapter9/description.c
@@ -53,6 +53,8 @@ cards_amd[] = {
     {"KAVERI",                      "AMD Radeon(TM) R7 Graphics"},
     {"KABINI",                      "AMD Radeon HD 8400 / R3 Series"},
     {"BONAIRE",                     "AMD Radeon HD 8770"},
+    {"LIVERPOOL",                   "AMD Liverpool (Playstation 4)"},
+    {"GLADIUS",                     "AMD Gladius (Playstation 4 Pro)"},
     {"OLAND",                       "AMD Radeon HD 8670"},
     {"HAINAN",                      "AMD Radeon HD 8600M Series"},
     {"TAHITI",                      "AMD Radeon HD 7900 Series"},
diff --git a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
index 7c3e99404f9..19b861a3f59 100644
--- a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
+++ b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
@@ -275,6 +275,8 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
       ws->info.chip_class = GFX6;
       break;
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_KABINI:
    case CHIP_HAWAII:
@@ -496,6 +498,8 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
    case CHIP_VERDE:
    case CHIP_OLAND:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.max_tcc_blocks = 4;
       break;
@@ -525,10 +529,12 @@ static bool do_winsys_init(struct radeon_drm_winsys *ws)
       case CHIP_TAHITI:
       case CHIP_PITCAIRN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
          ws->info.max_se = 2;
          break;
       case CHIP_HAWAII:
-         ws->info.max_se = 4;
+      case CHIP_GLADIUS:
+	 ws->info.max_se = 4;
          break;
       }
    }
